server/controllers/MiscController.js

```diff
@@ -258,6 +258,34 @@ class MiscController {
     })
     res.json(tags)
   }
+  // GET: api/social
+  async getSocialStats(req, res) {
+    if (this.db.serverSettings.sharedListeningStats == false) {
+      return res.sendStatus(404)
+    }
+    Logger.error(this.db.serverSettings.sharedListeningStats)
+    var filteredUsers = this.db.users.filter(c => c.settings.shareListeningActivity == true)
+    var userData = []
+    for (let i = 0; i < filteredUsers.length; i++) {
+      var user = filteredUsers[i]
+      var listeningStats = await this.getUserListeningStatsHelpers(user.id)
+      var latestItem = user.mediaProgress.sort(p => p.lastUpdate)
+      var totalItems = Object.keys(listeningStats.items).length
+      var totalDays = Object.keys(listeningStats.days).length
+      var session = {
+        username: user.username,
+        lastSeen: user.lastSeen,
+        latest: latestItem[latestItem.length - 1] || null,
+        itemsRead: totalItems || 0,
+        minutesListened: listeningStats.totalTime || 0,
+        daysListened: totalDays| 0,
+      }
```

Comment:
From what is being displayed in the UI it seems like we are pulling too much data here. For instance `itemsRead` and `daysListened` isn't being used.
From comment I left above I'm not sure `minutesListened` is what we want to display since it is total all time listening minutes.
It may make more sense to use the `getUserListeningSessionsHelper` instead or just use the `mediaProgress` array on the User object.

Reply:
I got rid of totalDays and totalItems, does the updated logic look better?